velox/connectors/hive/HiveConnector.cpp

```diff
@@ -80,29 +85,49 @@ std::string HiveTableHandle::toString() const {
 }
 
 HiveDataSink::HiveDataSink(
-    std::shared_ptr<const RowType> inputType,
-    const std::string& filePath,
-    velox::memory::MemoryPool* memoryPool)
-    : inputType_(inputType) {
+    RowTypePtr inputType,
+    std::shared_ptr<const HiveInsertTableHandle> insertTableHandle,
+    const ConnectorQueryCtx* FOLLY_NONNULL connectorQueryCtx)
+    : inputType_(std::move(inputType)),
+      insertTableHandle_(std::move(insertTableHandle)),
+      connectorQueryCtx_(connectorQueryCtx) {
+  if (!insertTableHandle_->isPartitioned()) {
+    writers_.emplace_back(createWriter());
+  }
+}
+
+void HiveDataSink::appendData(VectorPtr input) {
+  // For the time being the hive data sink supports one file.
+  // To extend it we can create a new writer for every
+  // partition.
+  if (writers_.empty()) {
+    writers_.emplace_back(createWriter());
```

Comment:
It seems that we always write use the first writer to write? Then why we make writers_ a vector in this PR? Thanks!

Reply:
This is for partition support in the coming PR. Otherwise, I need to resolve conflicts every time I rebase